server/src/test/java/com/hedera/block/server/persistence/storage/read/BlockAsLocalFileReaderTest.java

```diff
@@ -29,25 +51,64 @@
+    private PersistenceStorageConfig testConfig;
     private BlockAsLocalFileReader toTest;
 
+    @TempDir
+    private Path testLiveRootPath;
+
     @BeforeEach
-    void setUp() {
-        toTest = BlockAsLocalFileReader.newInstance();
+    void setUp() throws IOException {
+        blockNodeContext = TestConfigUtil.getTestBlockNodeContext(
+                Map.of(PERSISTENCE_STORAGE_LIVE_ROOT_PATH_KEY, testLiveRootPath.toString()));
+        testConfig = blockNodeContext.configuration().getConfigData(PersistenceStorageConfig.class);
+
+        final String testConfigLiveRootPath = testConfig.liveRootPath();
+        assertThat(testConfigLiveRootPath).isEqualTo(testLiveRootPath.toString());
+
+        blockAsLocalFileResolverMock = spy(BlockAsLocalFilePathResolver.of(testLiveRootPath));
+        blockAsLocalFileWriterMock = spy(BlockAsLocalFileWriter.of(blockNodeContext, blockAsLocalFileResolverMock));
+        toTest = BlockAsLocalFileReader.of(blockAsLocalFileResolverMock);
     }
 
     /**
      * This test aims to verify that the
      * {@link BlockAsLocalFileReader#read(long)} correctly reads a block with
      * a given block number.
-     *
-     * @param toRead parameterized, block number
      */
-    @ParameterizedTest
-    @MethodSource("validBlockNumbers")
-    void testSuccessfulBlockRead(final long toRead) {
-        // todo currently throws UnsupportedOperationException
-        assertThatExceptionOfType(UnsupportedOperationException.class).isThrownBy(() -> toTest.read(toRead));
+    @Test
+    void testSuccessfulBlockRead() throws IOException, ParseException {
+        final List<BlockItemUnparsed> blockItemUnparsed = PersistTestUtils.generateBlockItemsUnparsed(1);
+        final Optional<List<BlockItemUnparsed>> written = blockAsLocalFileWriterMock.write(blockItemUnparsed);
+
+        assertThat(written).isNotNull().isPresent();
+
+        final Optional<BlockUnparsed> actual = toTest.read(1);
+        assertThat(actual)
+                .isNotNull()
+                .isPresent()
+                .get(InstanceOfAssertFactories.type(BlockUnparsed.class))
+                .isNotNull()
+                .isExactlyInstanceOf(BlockUnparsed.class)
+                .returns(1L, from(blockUnparsed -> {
+                    try {
+                        return BlockHeader.PROTOBUF
+                                .parse(Objects.requireNonNull(
+                                        blockUnparsed.blockItems().getFirst().blockHeader()))
+                                .number();
+                    } catch (final ParseException e) {
+                        throw new RuntimeException(e);
+                    }
+                }))
+                .extracting(BlockUnparsed::blockItems)
+                .asList()
+                .isNotNull()
+                .isNotEmpty()
+                .hasSize(blockItemUnparsed.size())
+                .containsExactlyElementsOf(blockItemUnparsed);
```

Comment:
These two items are almost certainly overly specific and make the test brittle.
Verifying that what is read is _exactly_ what was written is better done in a separate test _by itself_ with its own input to reduce the probability of tests broken in ways that have nothing to do with whether the design is correct.

Reply:
✅ That is a good point, I split the tests. I do agree that the tests should be testing specific aspects of a given method and when there are such clearly defined and visible aspects of a method, like the method relies on something to be present on the filesystem and only then it can read the contents, then these are indeed two different tests, because if one breaks, this would very easily indicate which aspect of the intended functionality was broken.

However for the brittleness, while I do agree with you that it is good to have locally generated data, I have added another utility method to generate the test data so it could be reused. I think that if there is another case, where we need test data generated differently, we should add another method and not tamper with the logic of the already existing utility method that generates our data. For such a complex data generation logic, I do actually think it is better to indeed be in a publicly accessible utility, because otherwise if something changes with the data model, it will be a very big hustle to support.